### PR TITLE
Expanding info on the agent access token & registration token scopes

### PR DESCRIPTION
**Pull request by karensawrey**
Regarding the Issue #1413.

### CODE REVIEW

**Comment by karensawrey** [comment]
The PR is out of its draft stage but needs a fresh re-read before an actual human being will be kindly asked to give it a go.

File: pages/agent/v3/tokens.md.erb [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 
-The Buildkite Agent requires an agent token to connect to Buildkite and register for work. If you are an admin of your Buildkite organization, you can view the tokens on your [Agents page](https://buildkite.com/organizations/-/agents).
+The Buildkite agent requires an agent token to connect to Buildkite and register for work. If you are an admin of your Buildkite organization, you can view the tokens on your [Agents page](https://buildkite.com/organizations/-/agents).
 
 {:toc}
 
 ## The default token
 
-When you create a new organization in Buildkite, a default agent token is created. This token can be used for testing and development, but it's recommended to [create new, specific tokens](#creating-tokens) for each new environment.
+When you create a new organization in Buildkite, a default agent token (agent registration token) is created to start an agent. This token but cannot be used with the artifacts or meta-data. While it can be used for testing and development, it's recommended to [create new, specific tokens](#creating-tokens) for each new environment.
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:9]: > This token but cannot be used with the artifacts or meta-data.

this needs more words :-p

File: pages/agent/v3/tokens.md.erb [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+
+Agent access tokens (agent session tokens) are created when an agent is registered with Buildkite, and are unique for every registered agent.
+
+Agent access token only works as long as the agent that created it is running, and it is used by all agent operations once the agent is [started](/docs/agent/v3/cli-start).
+
+Access tokens are exposed within the job environment.
+
+## Exchanging tokens
+
+Agent registration token (default token) can be exchanged for an agent session token without any additional checks. To do it, the agent needs to call the `https://agent.buildkite.com/v3/register` endpoint with its registration token in the authorization header. The API response to this query is a regular agent access token that is next ingested and used by the agent.
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:23]: Does this happen automatically? Or do you need to save the agent access token

File: pages/agent/v3/tokens.md.erb [hunk 3]
```diff
@@ -1,12 +1,26 @@
 # Agent Tokens
 
-The Buildkite Agent requires an agent token to connect to Buildkite and register for work. If you are an admin of your Buildkite organization, you can view the tokens on your [Agents page](https://buildkite.com/organizations/-/agents).
+The Buildkite agent requires an agent token to connect to Buildkite and register for work. If you are an admin of your Buildkite organization, you can view the tokens on your [Agents page](https://buildkite.com/organizations/-/agents).
 
 {:toc}
 
 ## The default token
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:7]: Let's call this 

## Agent registration token (default token)

File: pages/agent/v3/tokens.md.erb [hunk 4]
```diff
@@ -97,6 +111,12 @@ Once a token is revoked, no new agents will be able to start with that token. Re
 
 Agent tokens are specific to each Buildkite organization, and can be used to register an agent with any [queue](/docs/agent/v3/queues). Agent tokens can not be shared between organizations.
 
+Agent tokens define the access permissions (`read` and/or `write`) for actions to be performed on all jobs and pipelines. This is necessary as Buildkite agents can run for extended periods of time. An agent might get launched to service one pipeline and then pick up work for another one after. This means that the access tokens for agents need the access to all pipelines and jobs.
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:114]: > Agent tokens define the access permissions

I'm not sure what this means either

File: pages/agent/v3/tokens.md.erb [hunk 5]
```diff
@@ -97,6 +111,12 @@ Once a token is revoked, no new agents will be able to start with that token. Re
...
+
+## Log security considerations
+
+Agent access token can allow writing to the stdout of a job as job log streaming happens on the `https://agent.buildkite.com/v3/jobs/chunks` agent API endpoint authenticated by the agent access token. If a bad actor gains access to your agent access token, they would be able to falsify new logs, but they wouldn’t be able to remove any existing logs as the endpoint is append-only.
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:118]:
```suggestion
Agent access token allow writing to the stdout of a job as job log streaming happens on the `https://agent.buildkite.com/v3/jobs/chunks` agent API endpoint authenticated by the agent access token. If a bad actor gains access to your agent access token, they would be able to falsify new logs, but they wouldn’t be able to remove any existing logs as the endpoint is append-only.
```

File: pages/agent/v3/tokens.md.erb [hunk 6]
```diff
@@ -97,6 +111,12 @@ Once a token is revoked, no new agents will be able to start with that token. Re
...
+
 ## Session tokens
```
Comment (plaindocs) [inline pages/agent/v3/tokens.md.erb:120]: This is an agent session token, right? as in

> Agent access tokens (agent session tokens) are created when an agent is registered with Buildkite, and are unique for every registered agent.

So  let's be consistent how we talk about them

**Pull request closed without merging**